src/Compilers/CSharp/Portable/Symbols/Source/SourceAssemblySymbol.cs

```diff
@@ -2860,6 +2863,41 @@ private static string DefaultValue(TypeSymbol type)
             return null;
         }
 
+        /// <summary>
+        /// Returns data decoded from Obsolete attribute or null if there is no Obsolete attribute.
+        /// This property returns ObsoleteAttributeData.Uninitialized if attribute arguments haven't been decoded yet.
+        /// </summary>
+        internal sealed override ObsoleteAttributeData? ObsoleteAttributeData
+        {
+            get
+            {
+                // [assembly: Experimental] may have been specified in the assembly or one of the modules
+                var lazySourceAttributesBag = _lazySourceAttributesBag;
+                if (lazySourceAttributesBag != null && lazySourceAttributesBag.IsDecodedWellKnownAttributeDataComputed)
+                {
+                    var data = (CommonAssemblyWellKnownAttributeData)lazySourceAttributesBag.DecodedWellKnownAttributeData;
+                    if (data?.ExperimentalAttributeData is { } experimentalAttributeData)
+                    {
+                        return experimentalAttributeData;
+                    }
+                }
+
+                var lazyNetModuleAttributesBag = _lazyNetModuleAttributesBag;
```

Comment:
>_lazyNetModuleAttributesBag [](http://example.com/codeflow?start=49&length=27)

Hmm, I'm surprised there is a `_lazyNetModuleAttributesBag` here rather than just deferring to the `ModuleSymbol`. That said, `_lazyNetModuleAttributesBag` wasn't added in this PR.